### DISCUSSION TITLE
Can this app work over shizuku?

### DISCUSSION THREAD

**Discussion by Ashwin-droid** [Mygod/VPNHotspot#341] [Q&A] [answered]
I just got a new phone an want to use this app but i have no plans of rooting in the next 3 support years of os updates so can this work over shizuku api

**Comment by Mygod** [comment]
Without root, no. With Shizuku *and* root, maybe.

@zacharee Any thoughts? 🤣

**Comment by zacharee** [comment, marked as answer]
I played around with adding Shizuku support. There's some network forwarding stuff that still needs root, so it'll make a VPN and everything but connections just won't go through.

**Comment by kayrus** [comment]
@zacharee when adb is started as root ("rooted debugging option" in "developer options", e.g. in lineageos), shizuku also gets root privileges. can you submit a PR to add shizuku support?
